authorisation/middleware.go

```diff
@@ -201,24 +196,18 @@ func (m PermissionCheckMiddleware) IdentityHealthCheck(ctx context.Context, stat
 
 // Parse token using returned Parser object
 func (m PermissionCheckMiddleware) Parse(token string) (*permsdk.EntityData, error) {
-	//	jwtParser, err := NewCognitoRSAParser(m.IdentityClient.JWTKeys)
-	//	if err != nil {
-	//		j, _ := json.Marshal(m.IdentityClient.JWTKeys)
-	//		fmt.Println(m.IdentityClient.JWTKeys)
-	//		return nil, errors.New(string(j))
-	//	}
 	return m.jwtParser.Parse(token)
 }
 
 // GetCollectionIdAttribute provides an implementation of GetAttributesFromRequest. Retrieves and returns
 // header 'Collection-Id' from the request if it exists, otherwise returns an empty map. Never returns an
 // error as the header is not mandatory
-func GetCollectionIdAttribute(req *http.Request) (map[string]string, error) {
+func GetCollectionIDAttribute(req *http.Request) (map[string]string, error) {
 	attributes := make(map[string]string, 0)
 
-	collectionIdAttribute, _ := headers.GetCollectionID(req)
-	if collectionIdAttribute != "" {
-		attributes[collectionIdAttributeKey] = collectionIdAttribute
+	collectionIDAttribute, _ := headers.GetCollectionID(req)
```

Comment:
(minor): whey are we ignoring the errors here ? if the collectionID does not exist in the header , I guess it would be good to return an error ?

Reply:
I think `GetCollectionID` returns ErrHeaderNotFound if the header is empty, but the func does not consider this an error case. I could check for error type instead of checking for empty string. This would be less confusing.

Reply:
In fact this was addressed in dp-api-clients-go version 2. I will upgrade this, as other files in the project are already using version 2 as well.